File: app/src/main/java/com/xiaolu/mylibrarytool/model/TextModel.java
package com.xiaolu.mylibrarytool.model;

import com.xiaolu.mylibrary.net.RequestBodyUtils;
import com.xiaolu.mylibrarytool.App;
import com.xiaolu.mylibrarytool.bean.BaseObjectBean;
import com.xiaolu.mylibrarytool.bean.LoginBean;
import com.xiaolu.mylibrarytool.contract.ITextContract;

import java.util.TreeMap;

import io.reactivex.Observable;

/**
 * @author xiaol
 * @describe
 * @date 2020/10/28  15:01
 * - generate by MvpAutoCodePlus plugin.
 */

public class TextModel implements ITextContract.Model {

    private TreeMap<String, Object> map;

    @Override
    public Observable<BaseObjectBean<LoginBean>> login(String loginName, String password, String loginType, String verifyCode) {
        map = new TreeMap<>();
        map.put("loginName", loginName);
        map.put("password", password);
        map.put("loginType", loginType);
        map.put("verifyCode", verifyCode);
        return App.getClient().login(RequestBodyUtils.setRequestBody_JSON(map));
    }
}

File: app/src/main/java/com/xiaolu/mylibrarytool/model/TextModel.kt
package com.xiaolu.mylibrarytool.model

import com.xiaolu.mylibrary.mvpbase.BaseModel
import com.xiaolu.mylibrary.mvpbase.IBaseModel
import com.xiaolu.mylibrary.net.RequestBodyUtils
import com.xiaolu.mylibrarytool.App
import com.xiaolu.mylibrarytool.bean.*
import com.xiaolu.mylibrarytool.contract.ITextContract
import kotlinx.coroutines.CoroutineScope

/**
 * @describe
 * @author  admin
 * @date 2021/8/23  14:13
 * 								 - generate by MvpAutoCodePlus plugin.
 */

class TextModel : BaseModel(), ITextContract.Model {
    override suspend fun getBanner(): BaseListBean<BannerBean> {
        return App.getClient().getBanner()
    }

    override suspend fun login(loginRequestBean: LoginRequestBean): BaseObjectBean<LoginBean> {
        return App.getClient()
            .login(RequestBodyUtils.setRequestBody_JSON(loginRequestBean.toString()))
    }

}

